include/fi_mem.h

```diff
@@ -146,18 +146,14 @@ do {								\
 	*(void **) local_p = (fs)->next;				\
 	(fs)->next = p;						\
 } while (0)
-#define smr_freestack_pop(fs) smr_freestack_pop_impl(fs,	\
+#define smr_freestack_pop(fs) smr_freestack_pop_impl(fs, fs->next,	\
 	(char **) fs + ((char **) (fs)->next - (char **) fs->base_addr))
 
-static inline void* smr_freestack_pop_impl(void *fs, void *fs_next)
+static inline void* smr_freestack_pop_impl(void *fs, void *next, void *local)
```

Comment:
Can you move local to a local variable rather than having the caller pass it in as a parameter?